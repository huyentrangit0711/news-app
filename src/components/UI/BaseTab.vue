<template>
	<div class="base-tab">
		<nav class="tabs flex flex-col sm:flex-row">
			<button
				v-for="(tabHeader, index) in tabHeaders"
				:key="index"
				:data-target="`panel-${index + 1}`"
				class="tab py-4 px-6 block hover:text-blue-500 focus:outline-none border-b-2 font-medium"
				:class="{ 'active-tab': activeTab === index }"
				@click="onActiveTab(index)"
			>
				{{ tabHeader }}
			</button>
		</nav>
		<div id="panels">
			<div
				class="panel-1 tab-content py-5"
				v-for="(tabContent, index) in tabContents"
				:class="{ active: activeTab === index, hidden: activeTab !== index }"
				:key="index"
				:id="`panel-${index + 1}`"
			>
				<component :is="tabContent"></component>
			</div>
		</div>
	</div>
</template>

<script>
import { mapActions, mapGetters } from 'vuex';
export default {
	name: 'BaseTab',
	props: {
		tabHeaders: {
			type: Array,
			require: true,
		},
		tabContents: {
			type: Array,
			require: true,
		},
	},
	computed: {
		...mapGetters(['activeTab']),
	},
	methods: {
		...mapActions(['updateActiveTabs']),
		onActiveTab(index) {
			this.updateActiveTabs({ value: index });
			if (this.$route.path !== '/news') {
				this.$router.push('/news');
			}
		},
	},
};
</script>

<style>
.active-tab {
	@apply text-blue-500 border-blue-500;
}
</style>
