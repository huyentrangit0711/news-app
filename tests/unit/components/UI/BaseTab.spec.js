import { shallowMount, createLocalVue } from '@vue/test-utils';
import BaseTab from '@/components/UI/BaseTab';
import Vuex from 'vuex';
const localVue = createLocalVue();

localVue.use(Vuex);
describe('BaseTab UI Component', () => {
	let wrapper;
	let actions;
	let store;
	let getters;
	let state;
	beforeEach(() => {
		actions = {
			updateActiveTabs: jest.fn(),
		};
		state = {
			activeTab: 2,
		};
		getters = {
			activeTab: () => jest.fn().mockReturnValue(2),
		};
		store = new Vuex.Store({
			actions,
			getters,
			state,
		});
		wrapper = shallowMount(BaseTab, {
			store,
			localVue,
			mocks: {
				$route: {
					path: '/news/2432523',
				},
				$router: {
					push: jest.fn(),
				},
			},
		});
	});
	afterEach(() => {});
	it('Component display correctly with active tab from store ', () => {
		wrapper.vm.onActiveTab(2);
		expect(actions.updateActiveTabs).toHaveBeenCalled();
		expect(wrapper.vm.$router.push).toHaveBeenCalled();
	});
	it('Update "activeTab" from store correctly ', async () => {
		await wrapper.vm.$nextTick();
		expect(wrapper.vm.activeTab()).toEqual(2);
	});
	it('Component render correctly ', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
