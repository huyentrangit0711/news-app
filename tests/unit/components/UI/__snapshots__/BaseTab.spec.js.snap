// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseTab UI Component Component render correctly  1`] = `
<div class="base-tab">
  <nav class="tabs flex flex-col sm:flex-row"></nav>
  <div id="panels"></div>
</div>
`;
